### PR TITLE
[Samples][Java] Fix disparities of 52.teams-messaging-extensions-search-auth-config sample

### PR DESCRIPTION
**Pull request by Batta32**
Related to microsoft/botbuilder-java1165

## Proposed Changes
We compared the documentation/code migration/behavior of the `52.teams-messaging-extensions-search-auth-config` sample between [Java](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/java_springboot/52.teams-messaging-extensions-search-auth-config) and [C#](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/52.teams-messaging-extensions-search-auth-config) and we found disparities and fixes that this PR includes.

- Add missing validations of properties
- Update README
- Add missing comments
- Clean-up autogenerated files/folder related to Eclipse (e.g. target/.classpath)
- Add `package-info.java` file

## Testing
_Teams Messaging Extensions Search Auth Config sample working as expected_
image

image

image

### CODE REVIEW

File: samples/java_springboot/52.teams-messaging-extensions-search-auth-config/README.md [hunk 1]
```diff
@@ -1,53 +1,52 @@
...
     - __*If you don't have an Azure account*__ you can use this [Bot Framework registration](https://docs.microsoft.com/en-us/microsoftteams/platform/bots/how-to/create-a-bot-for-teams#register-your-web-service-with-the-bot-framework)
-    
-1) [Add Authentication to your bot via Azure Bot Service](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=csharp)
-
-1) Update the `resources/application.properties` configuration for the bot to use the `MicrosoftAppId`, `MicrosfotAppPassword`, `ConnectionName`, and `SiteUrl` from the Bot Framework registration. (Note the App Password is referred to as the "client secret" in the azure portal and you can always create a new client secret anytime.)
-
-1) __*This step is specific to Teams.*__
+- Update the `resources/application.properties` configuration for the bot to use the `MicrosoftAppId`, `MicrosfotAppPassword`, `ConnectionName`, and `SiteUrl` from the Bot Framework registration. (Note the App Password is referred to as the "client secret" in the azure portal and you can always create a new client secret anytime.)
```
Comment (fran893) [inline samples/java_springboot/52.teams-messaging-extensions-search-auth-config/README.md]: Can we specify what value should have `SiteUrl` property?

File: samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import com.microsoft.bot.schema.*;
```
Comment (fran893) [inline samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java]: Check these imports, should be individuals imports

File: samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java [hunk 3]
```diff
@@ -44,35 +45,106 @@
...
+            TurnContext turnContext,
+            AppBasedLinkQuery query) {
+        String state = query.getState(); // Check the state value
+        TokenResponse tokenResponse = this.getTokenResponse(turnContext, state).join();
+        if (tokenResponse == null || StringUtils.isBlank(tokenResponse.getToken())) {
+            // There is no token, so the user has not signed in yet.
+
+            // Retrieve the OAuth Sign in Link to use in the MessagingExtensionResult Suggested Actions
+            UserTokenProvider userTokenClient = (UserTokenProvider) turnContext.getAdapter();
+            SignInResource resource = userTokenClient.getSignInResource(turnContext, this.connectionName).join();
```
Comment (fran893) [inline samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java:113]: Why we are adding these `join()` calls?

File: samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java [hunk 4]
```diff
@@ -176,65 +253,65 @@ public TeamsMessagingExtensionsSearchAuthConfigBot(
...
+            UserTokenProvider userTokenClient = (UserTokenProvider) turnContext.getAdapter();
+            SignInResource resource = userTokenClient.getSignInResource(turnContext, connectionName).join();
+            String signInLink = resource.getSignInLink();
+            MessagingExtensionResponse extensionResponse = new MessagingExtensionResponse();
+            CardAction cardAction = new CardAction();
+            cardAction.setType(ActionTypes.OPEN_URL);
+            cardAction.setValue(signInLink);
+            cardAction.setTitle("Bot Service OAuth");
 
-                    return tokenProvider.getOAuthSignInLink(turnContext, connectionName)
```
Comment (fran893) [inline samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java:191]: It is okay to remove this?

**Review by fran893 — CHANGES_REQUESTED** [review]

File: samples/java_springboot/52.teams-messaging-extensions-search-auth-config/README.md (hunk 1 above)
Comment (Batta32) [reply samples/java_springboot/52.teams-messaging-extensions-search-auth-config/README.md]: Done

File: samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java (hunk 2 above)
Comment (Batta32) [reply samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java]: Done

File: samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java (hunk 3 above)
Comment (Batta32) [reply samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java:113]: As we discussed, I replaced the `thenApply` and `thenCompose` to `join()` in order to be sync. I think that the best approach is to have the thenApply for sure, but all the scenarios in teams are different..

I want to keep it as is

File: samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java (hunk 4 above)
Comment (Batta32) [reply samples/java_springboot/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java:191]: Yes, this was deleted in C#

**Pull request closed without merging**